geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/server/AbstractInfoIntegrationTest.java

```diff
@@ -98,71 +152,130 @@ public void shouldReturnClusterEnabledProperty() {
   }
 
   @Test
-  public void shouldReturnEmptyStringGivenUnknownParameter() {
+  public void shouldReturnMemorySections_givenMemorySectionParameter() {
+    List<String> nonMemoryProperties = getAllProperties();
+    nonMemoryProperties.removeAll(MEMORY_PROPERTIES);
+
+    String actualResult = jedis.info("memory");
+
+    assertThat(actualResult).contains(MEMORY_PROPERTIES);
+    assertThat(actualResult).doesNotContain(nonMemoryProperties);
+  }
+
+  @Test
+  public void shouldReturnKeySpaceSections_givenKeySpaceKeyspaceParameter() {
+    List<String> nonKeyspaceProperties = getAllProperties();
+    nonKeyspaceProperties.removeAll(KEYSPACE_PROPERTIES);
+
+    jedis.set("key", "value");
+    String actualResult = jedis.info("keyspace");
+
+    assertThat(actualResult).contains(KEYSPACE_PROPERTIES);
+    assertThat(actualResult).doesNotContain(nonKeyspaceProperties);
+  }
+
+  @Test
+  public void shouldReturnReplicationSections_givenKeySpaceReplicationParameter() {
+    List<String> nonReplicationProperties = getAllProperties();
+    nonReplicationProperties.removeAll(REPLICATION_PROPERTIES);
+
+    String actualResult = jedis.info("replication");
+
+    assertThat(actualResult).contains(REPLICATION_PROPERTIES);
+    assertThat(actualResult).doesNotContain(nonReplicationProperties);
+  }
+
+  @Test
+  public void shouldReturnEmptyString_givenUnknownParameter() {
     String actualResult = jedis.info("nonesuch");
     assertThat(actualResult).isEqualTo("");
   }
 
-
   @Test
-  public void shouldReturnDefaultsGivenDefaultParameter() {
+  public void shouldReturnDefaults_givenDefaultParameter() {
+    jedis.set("key", "value");
     String actualResult = jedis.info("default");
-    assertThat(actualResult).contains(CLUSTER_PROPERTIES);
-    assertThat(actualResult).contains(SERVER_PROPERTIES);
-    assertThat(actualResult).contains(PERSISTENCE_PROPERTIES);
+    assertThat(actualResult)
+        .contains(getAllProperties());
   }
 
   @Test
-  public void shouldReturnDefaultsGivenAllParameter() {
+  public void shouldReturnDefaults_givenAllParameters() {
+    jedis.set("key", "value"); // make sure keyspace is there
     String actualResult = jedis.info("all");
-    assertThat(actualResult).contains(CLUSTER_PROPERTIES);
-    assertThat(actualResult).contains(SERVER_PROPERTIES);
-    assertThat(actualResult).contains(PERSISTENCE_PROPERTIES);
+    assertThat(actualResult)
+        .contains(getAllProperties());
+  }
+
+  private List<String> getAllProperties() {
+    List<String> allProperties = new ArrayList<>();
+
+    allProperties.addAll(STATS_PROPERTIES);
+    allProperties.addAll(KEYSPACE_PROPERTIES);
+    allProperties.addAll(MEMORY_PROPERTIES);
+    allProperties.addAll(CLIENTS_PROPERTIES);
+    allProperties.addAll(CLUSTER_PROPERTIES);
+    allProperties.addAll(PERSISTENCE_PROPERTIES);
+    allProperties.addAll(SERVER_PROPERTIES);
+    allProperties.addAll(REPLICATION_PROPERTIES);
+
+    return allProperties;
   }
```

Comment:
It seems like you could do away with this method and just have `allProperties` as a constant since the properties that are added to the list never change.